### ISSUE TITLE
[Bug]: failed setting highlight with color_icons = false option

### ISSUE THREAD

**Issue by rudotriton** [akinsho/bufferline.nvim#433]
### What happened?

Bufferline throws a `Failed setting BufferLineDevIconDefault  highlight, something isn't configured correctly: 'fg' is not a valid color` or some variation of this with a different highligh group. I may be wrong, but I suspect its the `ctermfg = not color_icons and "fg" or colors.get_color({` line in `ui.lua` as `"fg"` doesn't seem to be a valid value for `ctermfg` in `nvim_set_hl`.

### What did you expect to happen?

No errors

### Additional Information

...

### commit

latest

**Comment by rudotriton** [comment]
```lua
require("bufferline").setup({
  options = {
    color_icons = false,
  }
})
```

in which case `set_one` seems to call `nvim_set_hl` for something like `BufferLineDevIconDefaultSelected` with

```lua
{
  background = "NONE",
  ctermfg = "fg",
  default = true,
  foreground = "fg"
}
```

which in my case throws, since `"fg"` doesn't seem to be valid for `ctermfg`

**Comment by akinsho** [comment]
@rudotriton are you using a transparent terminal/neovim theme?

**Comment by akinsho** [comment]
Basically `cterm` can take a value of `fg` as can `gui` as long as the user is using a background colour, if you have set background to `NONE` then this will cause errors.

**Comment by rudotriton** [comment]
I'm not using a transparent theme

**Comment by akinsho** [comment]
@rudotriton can you try the PR I tagged you in please

**Comment by akinsho** [comment]
@rudotriton I've merged #439 into `dev` so going to close this out now. Let me know if there are any issues still whilst using `dev` (as I changed the solution again)
